### PR TITLE
Restore basisu mipmaps by reverting handle gltf binary's changes.

### LINKED ISSUE #72237: Textures using Basis Universal compression no longer generate mipmaps
### Godot version

v4.0.beta.custom_build [a43db5afa]

### System information

Windows 10, Forward+, NVidia RTX 3070 (516.94)

### Issue description

After recompiling the engine, a grid texture using Basis Universal compression in a test project became extremely noisy, having lost its mipmaps.

Opening a Basis Universal texture in the inspector shows that there are no mips, even if Generate Mipmaps is enabled:
image
editor_screenshot_2023-01-28T184045

### Steps to reproduce

1. Import a texture to a project (The noisier the better).
2. Open texture in the inspector, enable mipmaps, change the compression mode to Basis Universal, and re-import.

### Minimal reproduction project

[Basis Universal MipMapping.zip](https://github.com/godotengine/godot/files/10526818/Basis.Universal.MipMapping.zip)

### PR DESCRIPTION
**Pull request by fire**
Fixes: https://github.com/godotengine/godot/issues/72237

Referencing https://github.com/godotengine/godot/commit/39922d7167178bab9587aa00a7a108b93db6240e

### CODE REVIEW

**Comment by BastiaanOlij** [comment]
I'm not too informed of the ins and outs of the basic universal format, but it seems to make sense. If we can load multiple mip levels.

This code does seem to assume there will always be mip levels in the file, will this still work ok if there are no mip levels in the file being loaded and we'd thus want to fall back on generating mip levels after import?

The code for saving to basic universal seems to strip out mip levels suggesting mips are optional in the file format.

**Comment by fire** [comment]
Out of band conversation:

```
Previously basisu was storing mip maps as an array of basisu in the internal format for Texture.
The previous pr tried to move to storing it as one basisu image with mipmaps.
So I wanted to restore it to an array of basisu images.
Reduz is not sure if it makes sense to revert, your work makes sense and it generates smaller files, so it may be better to fix it eventually
Fixing keep compatibility intact either case.
So the decision we attempt to fix the basisu decompression and keep the approach of 1 image for a hotfix in 4.0.x
Yeah. So the decision we attempt to fix the basisu decompression and keep the approach of 1 image for a hotfix in 4.0.x
5:17 AM
fire alright that makes the most sense to me
If you need to change the format in 4.x, you can still support the old format with some limitations too.
fire Actually, maybe it would be better to revert and keep separate images because, when we eventually implement texture streaming, the single image will not work.
reduz please advise then
fire is the win from single image very considerable?
Well I've not seen it work since the I last worked on it. I worked with lyuma because it got stuck.
I think I used it because I wanted basisu from gltfpack to work with mipmaps
oh right, that comes as single image
so I guess then we should simply support both approaches
for CompressedTexture we use one, and for StreamedTexture we used another
What's the outcome of not fixing it 3 days before stable?
basisu is unusable for 3d purposes due to aliasing
most people don't use basisu
I think we can fix it after 4.0 is stable
```

tl;dr This is a 4.0.x hotfix.

**Comment by fire** [comment]
Close due to review from reduz and is superseded by https://github.com/godotengine/godot/issues/72237.
